### PR TITLE
Home page: table of books

### PR DESCRIPTION
**Pull request by AnonymousWalker**
image

This change is [Reviewable](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/740)

### CODE REVIEW

File: jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/HomePageViewModel.kt [hunk 1]
```diff
@@ -233,4 +235,12 @@ class HomePageViewModel : ViewModel() {
         project.modifiedTs = LocalDateTime.now()
         updateProjectUseCase.update(project).subscribe()
     }
+
+    fun loadBookList() {
```
Comment (AnonymousWalker) [inline jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/HomePageViewModel.kt]: for development only. Will change before merging

**Review by jsarabia — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/740)* status: 0 of 17 files reviewed, 3 unresolved discussions (waiting on @AnonymousWalker and @mXaln)

___
*[`jvm/controls/src/main/kotlin/org/wycliffeassociates/otter/jvm/controls/demo/ui/components/WorkbookOptionTableCell.kt` line 51 at r2](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/740#-NToryQ82iOQm_LWYl4m:-NToryQ9CwLtWrcwrW24:bsp1gka) ([raw file](https://github.com/Bible-Translation-Tools/Orature/blob/d4382317c009a824561c061b49bc7831c99da731/jvm/controls/src/main/kotlin/org/wycliffeassociates/otter/jvm/controls/demo/ui/components/WorkbookOptionTableCell.kt#L51)):*
> ```Kotlin
>         onOpenAction: () -> Unit,
>         onExportAction: () -> Unit,
>         onDeleteAction: () -> Unit
> ```

maybe this should be defined as a component (BookMenu or something), and the actions here should be instead firing events which are subscribed on in a viewmodel?
___
*[`jvm/controls/src/main/kotlin/org/wycliffeassociates/otter/jvm/controls/demo/ui/components/WorkbookTableRow.kt` line 20 at r2](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/740#-NTosTy49WTcNuDm5UEz:-NTosTy49WTcNuDm5UF-:bpe2yog) ([raw file](https://github.com/Bible-Translation-Tools/Orature/blob/d4382317c009a824561c061b49bc7831c99da731/jvm/controls/src/main/kotlin/org/wycliffeassociates/otter/jvm/controls/demo/ui/components/WorkbookTableRow.kt#L20)):*
> ```Kotlin
> 
>         setOnMouseClicked {
>             // clicking on a row opens workbook
> ```

maybe this is also an event?

**Review by AnonymousWalker — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/740)* status: 0 of 21 files reviewed, 3 unresolved discussions (waiting on @jsarabia and @mXaln)

___
*[`jvm/controls/src/main/kotlin/org/wycliffeassociates/otter/jvm/controls/demo/ui/components/WorkbookOptionTableCell.kt` line 51 at r2](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/740#-NToryQ82iOQm_LWYl4m:-NTstrGZEYljxGk6UxCm:beh2ob0) ([raw file](https://github.com/Bible-Translation-Tools/Orature/blob/d4382317c009a824561c061b49bc7831c99da731/jvm/controls/src/main/kotlin/org/wycliffeassociates/otter/jvm/controls/demo/ui/components/WorkbookOptionTableCell.kt#L51)):*
<details><summary><i>Previously, jsarabia (Joe) wrote…</i></summary><blockquote>

maybe this should be defined as a component (BookMenu or something), and the actions here should be instead firing events which are subscribed on in a viewmodel?
</blockquote></details>

I replaced with the event firing method. For the size of this, I would keep it as a function.
___
*[`jvm/controls/src/main/kotlin/org/wycliffeassociates/otter/jvm/controls/demo/ui/components/WorkbookTableRow.kt` line 20 at r2](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/740#-NTosTy49WTcNuDm5UEz:-NTsu4Vg4sa4BJl9cDEk:b-896fix) ([raw file](https://github.com/Bible-Translation-Tools/Orature/blob/d4382317c009a824561c061b49bc7831c99da731/jvm/controls/src/main/kotlin/org/wycliffeassociates/otter/jvm/controls/demo/ui/components/WorkbookTableRow.kt#L20)):*
<details><summary><i>Previously, jsarabia (Joe) wrote…</i></summary><blockquote>

maybe this is also an event?
</blockquote></details>

Done.

**Review by jsarabia — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/740)* status: 0 of 22 files reviewed, 7 unresolved discussions (waiting on @AnonymousWalker and @mXaln)

___
*[`common/src/main/kotlin/org/wycliffeassociates/otter/common/data/workbook/WorkbookStatus.kt` line 5 at r7](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/740#-NTtaBtq3ENZOZB0Wr0b:-NTtaBtq3ENZOZB0Wr0c:brdjnau) ([raw file](https://github.com/Bible-Translation-Tools/Orature/blob/0e41e8c9bc1de7661a75c050164409390eaa267c/common/src/main/kotlin/org/wycliffeassociates/otter/common/data/workbook/WorkbookStatus.kt#L5)):*
> ```Kotlin
> import java.time.LocalDateTime
> 
> data class WorkbookStatus(
> ```

What is the purpose of this class? (In regards to trying to evaluate the name of it)
___
*[`common/src/main/kotlin/org/wycliffeassociates/otter/common/data/workbook/WorkbookStatus.kt` line 10 at r7](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/740#-NTta3IQAm-6l-luKkUY:-NTta3IQAm-6l-luKkUZ:buvwdjp) ([raw file](https://github.com/Bible-Translation-Tools/Orature/blob/0e41e8c9bc1de7661a75c050164409390eaa267c/common/src/main/kotlin/org/wycliffeassociates/otter/common/data/workbook/WorkbookStatus.kt#L10)):*
> ```Kotlin
>     val label: String,
>     val progress: Double,
>     val modifiedTimestamp: LocalDateTime? = null,
> ```

Why would this be null?
___
*[`jvm/controls/src/main/kotlin/org/wycliffeassociates/otter/jvm/controls/demo/ui/components/WorkbookOptionTableCell.kt` line 51 at r2](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/740#-NToryQ82iOQm_LWYl4m:-NTtjcXLBU7HhKTb6Lbk:bjfs6d2) ([raw file](https://github.com/Bible-Translation-Tools/Orature/blob/d4382317c009a824561c061b49bc7831c99da731/jvm/controls/src/main/kotlin/org/wycliffeassociates/otter/jvm/controls/demo/ui/components/WorkbookOptionTableCell.kt#L51)):*
<details><summary><i>Previously, AnonymousWalker (Tony Tran) wrote…</i></summary><blockquote>

I replaced with the event firing method. For the size of this, I would keep it as a function.
</blockquote></details>

What does size have to do with anything? You're still allocating an object regardless. It being defined as a class does not require it to be defined in its own file (and I don't think it should be). But fundamentally, the popup menu IS a component. We tend to make classes for components, and functions are for type safe builders; it being a class is going to allow for better extension as things change, etc.
___
*[`jvm/controls/src/main/kotlin/org/wycliffeassociates/otter/jvm/controls/tableview/WorkBookTableView.kt` line 18 at r7](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/740#-NTtj5_11qSzJUbsTFGu:-NTtj5_11qSzJUbsTFGv:blytr1w) ([raw file](https://github.com/Bible-Translation-Tools/Orature/blob/0e41e8c9bc1de7661a75c050164409390eaa267c/jvm/controls/src/main/kotlin/org/wycliffeassociates/otter/jvm/controls/tableview/WorkBookTableView.kt#L18)):*
> ```Kotlin
> class WorkBookTableView(
>     books: ObservableList<WorkbookStatus>
> ) : VBox() {
> ```

Should this just extend TableView rather than vbox though? and put this in the vbox on the View?
___
*[`jvm/controls/src/main/kotlin/org/wycliffeassociates/otter/jvm/controls/tableview/WorkBookTableView.kt` line 63 at r7](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/740#-NTtjG_DCBceQAe2-xdu:-NTtjG_DCBceQAe2-xdv:b9vqi83) ([raw file](https://github.com/Bible-Translation-Tools/Orature/blob/0e41e8c9bc1de7661a75c050164409390eaa267c/jvm/controls/src/main/kotlin/org/wycliffeassociates/otter/jvm/controls/tableview/WorkBookTableView.kt#L63)):*
> ```Kotlin
>                 isReorderable = false
>                 isResizable = false
>                 isSortable = false
> ```

should this be in the css?
___
*[`jvm/controls/src/main/kotlin/org/wycliffeassociates/otter/jvm/controls/tableview/WorkBookTableView.kt` line 75 at r7](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/740#-NTtjIbQ51Kmm11e_PEG:-NTtjIbQ51Kmm11e_PEH:bj8blc6) ([raw file](https://github.com/Bible-Translation-Tools/Orature/blob/0e41e8c9bc1de7661a75c050164409390eaa267c/jvm/controls/src/main/kotlin/org/wycliffeassociates/otter/jvm/controls/tableview/WorkBookTableView.kt#L75)):*
> ```Kotlin
>                 isReorderable = false
>                 isResizable = false
>                 isSortable = false
> ```

same as above

**Review by AnonymousWalker — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/740)* status: 0 of 22 files reviewed, 6 unresolved discussions (waiting on @jsarabia and @mXaln)

___
*[`jvm/controls/src/main/kotlin/org/wycliffeassociates/otter/jvm/controls/demo/ui/components/WorkbookOptionTableCell.kt` line 51 at r2](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/740#-NToryQ82iOQm_LWYl4m:-NTxd1KpEUc_L3tn6u2I:b-896fix) ([raw file](https://github.com/Bible-Translation-Tools/Orature/blob/d4382317c009a824561c061b49bc7831c99da731/jvm/controls/src/main/kotlin/org/wycliffeassociates/otter/jvm/controls/demo/ui/components/WorkbookOptionTableCell.kt#L51)):*
<details><summary><i>Previously, jsarabia (Joe) wrote…</i></summary><blockquote>

What does size have to do with anything? You're still allocating an object regardless. It being defined as a class does not require it to be defined in its own file (and I don't think it should be). But fundamentally, the popup menu IS a component. We tend to make classes for components, and functions are for type safe builders; it being a class is going to allow for better extension as things change, etc.
</blockquote></details>

Done.
___
*[`jvm/controls/src/main/kotlin/org/wycliffeassociates/otter/jvm/controls/tableview/WorkBookTableView.kt` line 18 at r7](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/740#-NTtj5_11qSzJUbsTFGu:-NTxXxtl7rbtw-aWmDxs:b-896fix) ([raw file](https://github.com/Bible-Translation-Tools/Orature/blob/0e41e8c9bc1de7661a75c050164409390eaa267c/jvm/controls/src/main/kotlin/org/wycliffeassociates/otter/jvm/controls/tableview/WorkBookTableView.kt#L18)):*
<details><summary><i>Previously, jsarabia (Joe) wrote…</i></summary><blockquote>

Should this just extend TableView rather than vbox though? and put this in the vbox on the View?
</blockquote></details>

Done.
___
*[`jvm/controls/src/main/kotlin/org/wycliffeassociates/otter/jvm/controls/tableview/WorkBookTableView.kt` line 63 at r7](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/740#-NTtjG_DCBceQAe2-xdu:-NTxbn5Y1GDSJ6Bq4vJU:b-7xn219) ([raw file](https://github.com/Bible-Translation-Tools/Orature/blob/0e41e8c9bc1de7661a75c050164409390eaa267c/jvm/controls/src/main/kotlin/org/wycliffeassociates/otter/jvm/controls/tableview/WorkBookTableView.kt#L63)):*
<details><summary><i>Previously, jsarabia (Joe) wrote…</i></summary><blockquote>

should this be in the css?
</blockquote></details>

it may be in the css depending on whether it's supported. The css should be under workbookapp because it's specific to individual component.
___
*[`jvm/controls/src/main/kotlin/org/wycliffeassociates/otter/jvm/controls/tableview/WorkBookTableView.kt` line 75 at r7](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/740#-NTtjIbQ51Kmm11e_PEG:-NTxcyAL7g0mo4F5rXcU:b3ks9) ([raw file](https://github.com/Bible-Translation-Tools/Orature/blob/0e41e8c9bc1de7661a75c050164409390eaa267c/jvm/controls/src/main/kotlin/org/wycliffeassociates/otter/jvm/controls/tableview/WorkBookTableView.kt#L75)):*
<details><summary><i>Previously, jsarabia (Joe) wrote…</i></summary><blockquote>

same as above
</blockquote></details>

^

**Review by AnonymousWalker — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/740)* status: 0 of 23 files reviewed, 6 unresolved discussions (waiting on @jsarabia and @mXaln)

___
*[`common/src/main/kotlin/org/wycliffeassociates/otter/common/data/workbook/WorkbookStatus.kt` line 5 at r7](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/740#-NTtaBtq3ENZOZB0Wr0b:-NTxv54A6_6FzWcaMrsG:b-896fix) ([raw file](https://github.com/Bible-Translation-Tools/Orature/blob/0e41e8c9bc1de7661a75c050164409390eaa267c/common/src/main/kotlin/org/wycliffeassociates/otter/common/data/workbook/WorkbookStatus.kt#L5)):*
<details><summary><i>Previously, jsarabia (Joe) wrote…</i></summary><blockquote>

What is the purpose of this class? (In regards to trying to evaluate the name of it)
</blockquote></details>

Done.
___
*[`common/src/main/kotlin/org/wycliffeassociates/otter/common/data/workbook/WorkbookStatus.kt` line 10 at r7](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/740#-NTta3IQAm-6l-luKkUY:-NTxv5JR6S4kJEj7P0Zx:b-896fix) ([raw file](https://github.com/Bible-Translation-Tools/Orature/blob/0e41e8c9bc1de7661a75c050164409390eaa267c/common/src/main/kotlin/org/wycliffeassociates/otter/common/data/workbook/WorkbookStatus.kt#L10)):*
<details><summary><i>Previously, jsarabia (Joe) wrote…</i></summary><blockquote>

Why would this be null?
</blockquote></details>

Done.

**Review by jsarabia — COMMENTED** [review]
> Reviewed 3 of 14 files at r2, 1 of 14 files at r3, 1 of 4 files at r5, 1 of 4 files at r7, 2 of 8 files at r9, 1 of 2 files at r11, 1 of 2 files at r12.
> *[Reviewable](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/740)* status: 10 of 24 files reviewed, 2 unresolved discussions (waiting on @mXaln)

**Review by jsarabia — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/740#-:-NU1d9Al1JCmUboH-P80:bnfp4nl)* status: 10 of 24 files reviewed, 2 unresolved discussions (waiting on @mXaln)

**Pull request merged**